janus_core/src/test_util/kubernetes.rs

```diff
@@ -0,0 +1,60 @@
+//! Testing framework for functionality that interacts with Kubernetes.
+
+use kube::config::KubeConfigOptions;
+use rand::{thread_rng, Rng};
+use std::process::{Command, Stdio};
+
+/// EphemeralCluster represents a running ephemeral Kubernetes cluster for testing. Dropping an
+/// EphemeralCluster will cause the associated Kubernetes cluster to be stopped & cleaned up.
+pub struct EphemeralCluster {
+    name: String,
+}
+
+impl EphemeralCluster {
+    /// Creates & starts a new ephemeral Kubernetes cluster.
+    pub fn create() -> Self {
+        // Choose a cluster name.
+        let mut randomness = [0u8; 4];
+        thread_rng().fill(&mut randomness);
+        let cluster_name = format!("janus-ephemeral-{}", hex::encode(&randomness));
+
+        // Use kind to start the cluster.
+        assert!(Command::new("kind")
+            .args(["create", "cluster", "--name", &cluster_name])
```

Comment:
I'd recommend that we use a separate temporary kubecfg file for this, rather than having test commands touch the default one. One big benefit is that this would avoid different commands trying to concurrently modify the file when running multiple tests. `kind` takes a `--kubeconfig` argument, as does `kubectl`.